### ISSUE TITLE
I can't share data between Splade and Vue Components

### ISSUE THREAD

**Issue by seewhy17** [protonemedia/laravel-splade#448]
- Laravel Version: 10.13.5
- PHP Version: 8.1.13
- Splade JS Version (npm): 1.4.11
- Splade PHP Version (composer): 1.4.11
- Dev environment (OS, Sail/Valet/etc): Windows 10 Pro, WAMP

### Description:

I am working on a project with a handful of UI interactions that relies on reactive properties.
### 1. First, how do I get the ```@scroll``` event to work? I could not either within Vue or Splade components. 
so I resulted to the following approach:

```Guest.vue```
```vue
<script setup>
    import { reactive, onMounted, inject } from 'vue'
    import { throttle } from 'lodash-es'

    const Splade = inject('$splade')

    const page= reactive({
        yScrollPosition: 0,
        onScroll: throttle(() => {
            state.yScrollPosition = window.scrollY
            Splade.emit('page-scroll', { y: state.yScrollPosition })

        }, 200),
    })

    onMounted(() => {
        window.addEventListener('scroll', page.onScroll)
    })
</script>
<template>
    <div>
        <slot />
    </div>
</template>

```

  
While this works, I haven't been able to update the Splade components (nested or otherwise) that require the payload of the emitted event.

### Steps To Reproduce Issue:
The ```Guest.vue``` component wraps a header, main and footer layout so

```
...
    <Guest v-bind="$attrs">
        <div>
            @include('layouts.nav')
            <main>
                {{ $slot }}
            </main>
            @include('layouts.footer')
        </div>

    </Guest>
...    
    
``` 

### 3. How do I update a parent's reactive property or store reactive property from the ```<x-splade-script>``` component?

I have tried to retrieve the emitted payload but could not bind it to a reactive property on the receiving component

```
<x-splade-script>
    $splade.on('page-scroll', function(payload) {
        console.log(payload.y)
        // updating a default data or store fails
        yPosition = payload.y // (local) this fails
        page.yPosition = payload.y // (store) this also fails

    });
</x-splade-script>

```

### 5. Is there a way to listen for events fired via the event bus on Splade Components?

In short, all my attempts to share data between Vue and Splade have not been productive.

**_Converting all the components to Vue components and resulting back to emitting upwards or props downward is not the solution am looking for._**

@pascalbaljet Thanks for this amazing Package.

**Comment by seewhy17** [comment]
The No 1 issue is not a problem, I'm just so used to the Alpine.js way of doing things

**Comment by seewhy17** [comment]
I guess these questions actually made things a little bit complicated to answer.
What I wanted was to share reactive states across my layout and having the ```default``` attribute of ```<x-splade-data>``` hold that state was a bit too restricting, so was using a regular Vue component.

I opted for a renderless vue component and that gave me access to all the state I needed

```vue

<script setup>
    import { ref, reactive, computed, onMounted, onUnmounted, watch } from 'vue'
    import { debounce, throttle } from 'lodash-es'


    const Guest = reactive({
        currentURL: null,
        faqReference: null,
        footerHeight: 0,
        mobileMenu: null,
        navbarHeight: 0,
        openMobileMenu: false,
        onFaq: false,
        yScrollPosition: 0,
        minMainHeight: null,
        headerFooterHeights: computed(() => Guest.navbarHeight + Guest.footerHeight),
        onScroll:throttle(() => {
            Guest.yScrollPosition = window.scrollY
        }, 60),
        scrollFaqIntoView:debounce(() => {
            Guest.faqReference.scrollIntoView()
        }, 200),

        isFaq() {
            const url = new URL(window.location.href)
            if (url.hash === '#faq') {
                Guest.onFaq = true
                Guest.openMobileMenu = false
                Guest.scrollFaqIntoView()
            }
        },

        toggleMobileMenu() {
            Guest.openMobileMenu = !Guest.openMobileMenu
        },

        closeMobileMenu(event) {
            Guest.openMobileMenu = event.target.matches('#mobileMenu')
        }

    })


    watch(
        () => Guest.currentURL,
        () => Guest.isFaq()
    )

    onMounted(() => {
        Guest.currentURL = window.history.state.url
        Guest.faqReference = document.querySelector('#faq')
        Guest.mobileMenu = document.querySelector('#mobileMenu')
        window.addEventListener('scroll', Guest.onScroll)
    })

    onUnmounted(() => {
        window.removeEventListener('scroll', Guest.onScroll)
    })

</script>
<template>
    <slot
        :closeMobileMenu="Guest.closeMobileMenu"
        :footerHeight="Guest.footerHeight"
        :minMainHeight="Guest.minMainHeight"
        :navbarHeight="Guest.navbarHeight"
        :openMobileMenu="Guest.openMobileMenu"
        :onFaq="Guest.onFaq"
        :toggleMobileMenu="Guest.toggleMobileMenu"
        :yScrollPosition="Guest.yScrollPosition"

    />
</template>

```

### Usage
```
<Guest
    v-slot="{
        closeMobileMenu,
        footerHeight,
        minMainHeight,
        navbarHeight,
        openMobileMenu,
        onFaq,
        toggleMobileMenu,
        yScrollPosition
    }"
>
    <div
        v-bind="$attrs"
        class="min-h-screen grid grid-rows-[auto_1fr_auto] relative z-0"
        @click.stop="closeMobileMenu"
    >
        @include('layouts.nav')
        <main id="main">
            {{ $slot }}
        </main>
        @include('layouts.footer')
        <x-kit.scrolltotop />

    </div>
</Guest>

```

@pascalbaljet thanks again for this amazing package!
